pkg/cluster/prepare_test.go

```diff
@@ -0,0 +1,325 @@
+package cluster_test
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/weaveworks/pctl/pkg/cluster"
+	"github.com/weaveworks/pctl/pkg/runner/fakes"
+)
+
+type mockTransport struct {
+	res *http.Response
+}
+
+func (t *mockTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	return t.res, nil
+}
+
+var _ = Describe("prepare", func() {
+	When("dry run is set", func() {
+		It("can prepare the environment with everything that profiles needs without actually modifying the cluster", func() {
+			runner := &fakes.FakeRunner{}
+			content, err := ioutil.ReadFile(filepath.Join("testdata", "prepare.yaml"))
+			Expect(err).NotTo(HaveOccurred())
+			client := &http.Client{
+				Transport: &mockTransport{
+					res: &http.Response{
+						StatusCode: http.StatusOK,
+						Body:       ioutil.NopCloser(bytes.NewReader(content)),
+					},
+				},
+			}
+			tmp, err := ioutil.TempDir("", "prepare_01")
+			Expect(err).NotTo(HaveOccurred())
+			p := &cluster.Preparer{
+				PrepConfig: cluster.PrepConfig{
+					Location: tmp,
+					DryRun:   true,
+				},
+				Fetcher: &cluster.Fetcher{
+					Client: client,
+				},
+				Applier: &cluster.Applier{
+					Runner: runner,
+				},
+			}
+			err = p.Prepare()
+			Expect(err).NotTo(HaveOccurred())
+			Expect(runner.RunCallCount()).To(Equal(1))
+			arg, args := runner.RunArgsForCall(0)
+			Expect(arg).To(Equal("kubectl"))
+			Expect(args).To(Equal([]string{"apply", "-f", filepath.Join(tmp, "prepare.yaml"), "--dry-run=client", "--output=yaml"}))
```

Comment:
worth testing that `prepare.yaml` contains the correct `contents`?

Reply:
It won't match since kube already applied some things like status, variables, a bunch of things that aren't in the original. So it's not like an `Equal` and that's that. I can compare individual things if I create a map yaml out of it... But that felt like overkill. :) Especially considering that we are testing the apply eventually in the integration tests.